### DISCUSSION TITLE
Plugin that adds a dependency to a module

### DISCUSSION THREAD

**Discussion by aomarks** [webpack/webpack#15998] [Q&A]
I would like to write a Webpack 5 plugin which, whenever a specific module is imported (a third-party module that was installed to node_modules), another specific module (a different third-party module in node_modules) is loaded first. Or in other words, I would like to effectively inject an import of a module into a module.

One way I thought of for doing this is to use the normalModuleFactory resolve (or beforeResolve?) hook, and push a new dependency into the dependencies array. However, I haven't been able to work out how to construct a dependency object that would have the effect I am looking for.

Does anybody have any experience or advice here?

Thanks!

**Comment by vankop** [comment]
better to use loader for this, basically you want modify module source code.
